Dockerfile

```diff
@@ -16,61 +25,82 @@ RUN turbo prune --scope=@kampus-apps/gql --docker
 FROM base AS pasaport-prune
 RUN turbo prune --scope=@kampus-apps/pasaport --docker
 
-FROM node:18-bullseye-slim as kampus-builder
+# kampus-builder stage, use "turbo" to build the project incrementally and get incremental builds
+FROM node-slim-image as kampus-builder
 WORKDIR /app
 COPY .gitignore .gitignore
 COPY --from=kampus-prune /app/out/json .
 COPY --from=kampus-prune /app/out/package-lock.json ./package-lock.json
 COPY /tsconfig.json tsconfig.json
 RUN npm install
 COPY --from=kampus-prune /app/out/full/ .
+COPY .env ./apps/pasaport/.env
 COPY turbo.json turbo.json
 RUN npx turbo build
 
 # python, g++, make, etc. are required for gql dependencies. node:18 is used as base image
-FROM node:18 AS gql-builder
+FROM node-image AS gql-builder
 WORKDIR /app
 COPY .gitignore .gitignore
 COPY --from=gql-prune /app/out/json .
 COPY --from=gql-prune /app/out/package-lock.json ./package-lock.json
 COPY /tsconfig.json tsconfig.json
 RUN npm install
 COPY --from=gql-prune /app/out/full/ .
+COPY --from=gql-prune /app/out/full/apps/gql/.env.example ./apps/gql/.env
+COPY .env ./apps/pasaport/.env
```

Comment:
@usirin should we make a small script to do this on local as well seems like right now .env.example files can run the applications so local env can utilise this too.

Reply:
from what i read from the next.js docs, the best practice is to switch to a `.env.development` file and commit that. then people can create a `.env.local` file if they want to override anything from that files.

> Good to know: .env, .env.development, and .env.production files should be included in your repository as they define defaults. .env*.local should be added to .gitignore, as those files are intended to be ignored. .env.local is where secrets can be stored.
> https://nextjs.org/docs/app/building-your-application/configuring/environment-variables#default-environment-variables